dbms/src/Storages/getStructureOfRemoteTable.cpp

```diff
@@ -31,9 +32,42 @@ ColumnsDescription getStructureOfRemoteTable(
     const Context & context,
     const ASTPtr & table_func_ptr)
 {
-    /// Send to the first any remote shard.
-    const auto & shard_info = cluster.getAnyShardInfo();
+    const auto & shards_info = cluster.getShardsInfo();
 
+    std::string fail_messages;
+
+    for (auto & shard_info : shards_info)
+    {
+        try {
+            const auto & res = getStructureOfRemoteTableInShard(shard_info, database, table, context, table_func_ptr);
+            return res;
+        }
+        catch (const DB::NetException & e)
+        {
+            /// THIS DOES NOT WORK AND IS ALWAYS FALSE???
+            if (context.getSettingsRef().skip_unavailable_shards)
+            {
+                std::string fail_message = getCurrentExceptionMessage(false);
+                fail_messages += fail_message + '\n';
```

Comment:
Would be nice to log these. How to get the logger in this context, just pass it to the function?

Reply:
It's `LOG_INFO(&Poco::Logger::get("<some arbitrary name of your logging scope>"), "<the message>")`.

Going to improve this soon because it's absolutely undiscoverable, good that you asked.